src/spac/transformations.py

```diff
@@ -493,3 +493,45 @@ def normalize_features(
     adata.layers[new_layer_name] = dataframe
 
     return quantiles
+
+
+def z_score_normalization(adata, layer=None):
+    """
+    Compute z-scores for the provided AnnData object.
+
+    Parameters
+    ----------
+    adata : anndata.AnnData
+        The AnnData object containing the data to normalize.
+    layer : str, optional
+        The name of the layer in the AnnData object to normalize.
+        If None, the main data matrix .X is used.
+
+    Returns
+    -------
+    adata : anndata.AnnData
+        The AnnData object with a new layer 'z_scores' containing the computed
+        z-scores.
+    """
+
+    # Check if the provided layer exists in the AnnData object
+    if layer:
+        check_table(adata, tables=layer)
+
+        means = np.mean(adata.layers[layer], axis=0)
+        std_devs = np.std(adata.layers[layer], axis=0)

```

Comment:
Thanks @fangliu117, minor though: instead of adding this line twice, the numpy array first, then write the calculation of the Z score one. You can use this function for the z score if we would need later to use any of the options: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.zscore.html

Reply:
Thanks @georgezakinih for letting me know this scipy.stats.zscore. It comes with additional functionalities, such as handling NaN values and providing degrees of freedom corrections, which will be beneficial in the future.

So the manual computation of the z-scores in my original function is redundant, then I transitioned to using scipy.stats.zscore to refactor the function.